src/app/browse/br-keyword/br-keyword.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, OnInit } from '@angular/core';
+import {Router, ActivatedRoute} from '@angular/router';
+import {Location} from '@angular/common';
+import { item } from '../../Item';
+import { ItemService } from '../../item.service';
+import { Mock } from '../../MockBook';
+
+@Component({
+  selector: 'app-br-keyword',
+  templateUrl: './br-keyword.component.html',
+  styleUrls: ['./br-keyword.component.css']
+})
+export class BrKeywordComponent implements OnInit {
+  retrievedItems: item[];
+  items: item[];
+  keywords: string[];
+
+  constructor(public route: ActivatedRoute, 
+             public router: Router,
+             public ItemService: ItemService) {}
+
+
+  getKeywordItems(retrievedItems:item[]):void{
+    var acceptedItems: number[] = [];
+
+    console.log(retrievedItems)
+    for (const index in retrievedItems){
+      var item:item = retrievedItems[index]
+//        console.log(item)
+        if (this.keywords.every(keyword => this.keywordCheck(keyword,item))){
+          acceptedItems.push(parseInt(index))
+        }
+    }
+//    console.log(acceptedItems)
+
+    this.items = [];
+    for (const index in retrievedItems){
+      if (acceptedItems.includes(parseInt(index))){
+        this.items.push(retrievedItems[index])
+      }
+    }
+
+//    console.log(this.items)
+  }
+ 
+  keywordCheck(keyword:string,item:item):boolean{
+  	if (item.description.toLowerCase().includes(keyword)){
+  		return true
+  	}
+  	return false
+  }
+
+  initializeKeywordBar():HTMLInputElement{
+  	var router = this.router
+  	var keywordBar = <HTMLInputElement>document.getElementById('keyword')
+
+  	keywordBar.addEventListener("keyup",function(event){
+  		if (event.keyCode == 13){
+  			event.preventDefault();
+  			router.navigate(['br-keyword/',keywordBar.value])
+  			
```

Comment:
Check out some of the features that Angular forms provide. This may simplify the process of listening to form submission/key-enter events.

- [Form Field](https://material.angular.io/components/form-field/overview)
- [Angular Material Autocomplete](https://material.angular.io/components/autocomplete/overview)

Also, you have some of this logic already but [this link](https://reactgo.com/angular-submit-form-enter/) might be helpful to use HTML `<form>` tags to take advantage of additional Angular features.